src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Diagnostics/DocumentPullDiagnosticsEndpoint.cs

```diff
@@ -106,6 +113,22 @@ public TextDocumentIdentifier GetTextDocumentIdentifier(VSInternalDocumentDiagno
         return allDiagnostics.ToArray();
     }
 
+    private IDisposable? Track(string name, Guid correlationId)
+    {
+        if (!System.Diagnostics.Debugger.IsAttached)
+        {
+            System.Diagnostics.Debugger.Launch();
+            System.Diagnostics.Debugger.Break();
+        }
+
+        return _telemetryReporter?.BeginBlock(name, Severity.Normal, ImmutableDictionary.CreateRange(new KeyValuePair<string, object?>[]
+        {
+            new("eventscope.method", "textdocument/_vs_diagnostic"),
+            new("eventscope.languageservername", "Razor Language Server"),
```

Comment:
I'm not sure if there is an existing constant for this anywhere, but if now should probably add one to `LanguageServerConstants` and use it instead of a hardcoded string.